dev/com.ibm.ws.webcontainer.servlet.3.1/src/com/ibm/ws/webcontainer31/upgrade/H2HandlerImpl.java

```diff
@@ -33,30 +34,33 @@
 public class H2HandlerImpl implements H2Handler {
 
     private static final TraceComponent tc = Tr.register(H2HandlerImpl.class,null);
-
+    private static final String CONSTANT_upgrade = "upgrade";
+    private static final String CONSTANT_connection = "connection";
+    private static final String HTTP2_SETTINGS = "HTTP2-Settings";
+    
     /**
      * Determines if a given request is an http2 upgrade request
      */
     @Override
     public boolean isH2Request(HttpInboundConnection hic, ServletRequest request) throws ServletException {
         
         //first check if H2 is enabled for this channel/port
-        if (!((Http2InboundConnection)hic).isHTTP2UpgradeRequest(null, true) ){
+        if (!((Http2InboundConnection) hic).isHTTP2UpgradeRequest(null, true)) {
             return false;
         }
-        
-        Map<String, String> headers = new HashMap<String, String>();
-        HttpServletRequest hsrt = (HttpServletRequest) request;
-        Enumeration<String> headerNames = hsrt.getHeaderNames();
-        
-        // create a map of the headers to pass to the transport code
+        // Retrieve the needed header values for this request
+        Map<String, String> h2Headers = null; 
+        Enumeration<String> headerNames = ((HttpServletRequest) request).getHeaderNames();
```

Comment:
((HttpServletRequest) request) happens in many places after this.  Any advantage of that VS saved the casting here?